### PR TITLE
added filters for course

### PR DESCRIPTION
**Pull request by CalvinFronda**
Adding filters for courses under the Registration tab

### CODE REVIEW

File: src/components/FeatureViews/Registration/RegistrationLanding.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import * as hooks from "actions/hooks";
-import React, {useCallback, useMemo, useState} from "react";
+import { distinctObjectArray, gradeOptions } from "../../../utils";
```
Comment (NeilAllavarpu) [inline src/components/FeatureViews/Registration/RegistrationLanding.js]: You can also use file paths relative to the `src/` directory for imports:

    import {...} from "utils"; // utils is at the top level inside src

File: src/components/FeatureViews/Registration/RegistrationLanding.js [hunk 2]
```diff
@@ -16,19 +17,31 @@ import Loading from "../../Loading";
 import CourseList from "./CourseList";
 import TutoringList from "./TutoringList";
 import RegistrationActions from "./RegistrationActions";
+import * as adminActions from "../../../actions/adminActions";
```
Comment (NeilAllavarpu) [inline src/components/FeatureViews/Registration/RegistrationLanding.js]: See above

File: src/components/FeatureViews/Registration/RegistrationLanding.js [hunk 3]
```diff
@@ -16,19 +17,31 @@ import Loading from "../../Loading";
...
+    const api = useMemo(
+        () => ({
+            ...bindActionCreators(adminActions, dispatch),
+        }),
+        [dispatch]
+    );
+
+    useEffect(() => {
+        api.fetchCategories();
+    }, [api]);
```
Comment (NeilAllavarpu) [inline src/components/FeatureViews/Registration/RegistrationLanding.js]: Replace this with the `useCategory` hook (imported as `hooks`).

File: src/components/FeatureViews/Registration/RegistrationLanding.js [hunk 4]
```diff
@@ -16,19 +17,31 @@ import Loading from "../../Loading";
...
-    hooks.useInstructor();
+    hooks.useInstructor()
+
+    const dispatch = useDispatch();
+    const api = useMemo(
+        () => ({
+            ...bindActionCreators(adminActions, dispatch),
+        }),
+        [dispatch]
+    );
```
Comment (NeilAllavarpu) [inline src/components/FeatureViews/Registration/RegistrationLanding.js]: If you use the `useCategory` hook instead, this is unnecessary.

File: src/components/FeatureViews/Registration/RegistrationLanding.js [hunk 5]
```diff
@@ -42,26 +55,32 @@ const RegistrationLanding = () => {
...
+        .map(({ name, user_id }) => ({
             "label": name,
             "value": user_id,
         })), [instructors]);
 
+    const subjectOptions = useMemo(() => distinctObjectArray(Object.values(courses)
+        .map(({ course_id, category }) => ({
+            "label": categories.find(Category => Number(Category.id) === Number(category)).name,
+            "value": category,
+        }))), [courses])
```
Comment (NeilAllavarpu) [inline src/components/FeatureViews/Registration/RegistrationLanding.js]: Why not simply pull categories from redux directly?

File: src/components/FeatureViews/Registration/RegistrationLanding.js (hunk 3 above)
Comment (CalvinFronda) [reply src/components/FeatureViews/Registration/RegistrationLanding.js]: Please explain. 
Screen Shot 2020-03-09 at 10 13 52 AM
error:
Screen Shot 2020-03-09 at 10 14 27 AM

File: src/components/FeatureViews/Registration/RegistrationLanding.js (hunk 5 above)
Comment (CalvinFronda) [reply src/components/FeatureViews/Registration/RegistrationLanding.js]: We need to match categories. Inside redux the category value is an integer that needs to match with the category object

**Comment by NeilAllavarpu** [comment]
Fix the merge conflict and it's good to be merged

**Pull request merged**